### PR TITLE
Add support to run container in cloud scripts

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
### Description

This PR adds supports to launch AWS or Azure with container.  The NVFlare server or client is running inside that container.  These four combinations are supported: Server+AWS, Server+Azure, Client+AWS and Client+Azure.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/lighter/impl/master_template.yml [hunk 1]
```diff
@@ -951,6 +961,22 @@ azure_start_svr_sh: |
...
   fi
 
+  if [ -z ${image_name+x} ]
+  then
+      container=false
+  else
+      container=true
+  fi
+
+  if [ $container = true ]
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/lighter/impl/master_template.yml:971]: Not necessary.  This syntax also work.

File: nvflare/lighter/impl/master_template.yml [hunk 2]
```diff
@@ -951,6 +961,22 @@ azure_start_svr_sh: |
...
+  then
+    VM_IMAGE=Canonical:0001-com-ubuntu-server-focal:20_04-lts-gen2:latest
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/lighter/impl/master_template.yml:973]: This is the default.  Users can change by either option or prompt.

File: nvflare/lighter/impl/master_template.yml [hunk 3]
```diff
@@ -1659,6 +1767,24 @@ aws_start_svr_sh: |
...
+  if [ -z ${image_name+x} ]
+  then
+      container=false
+  else
+      container=true
+  fi
+
+  if [ $container = true ]
+  then
+    AMI_IMAGE=ami-06b8d5099f3a8d79d
```
[the remark this answers is not in the document]
Comment (IsaacYangSLA) [reply nvflare/lighter/impl/master_template.yml:1779]: As the above.  This is the default.  Users can change by option or prompt.

**Review by nvidianz — APPROVED** [review]

**Pull request merged**